diff --git a/Makefile.in b/Makefile.in
index 3f554de..b8d1d5f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -242,12 +242,12 @@ libver_a_LIBADD =
 nodist_libver_a_OBJECTS = version.$(OBJEXT)
 libver_a_OBJECTS = $(nodist_libver_a_OBJECTS)
 am__gzip_SOURCES_DIST = bits.c deflate.c gzip.c inflate.c trees.c \
-	unlzh.c unlzw.c unpack.c unzip.c util.c zip.c dfltcc.c
+	unlzh.c unlzw.c unpack.c unzip.c util.c zip.c zedc.c dfltcc.c
 @IBM_Z_DFLTCC_TRUE@am__objects_1 = dfltcc.$(OBJEXT)
 am_gzip_OBJECTS = bits.$(OBJEXT) deflate.$(OBJEXT) gzip.$(OBJEXT) \
 	inflate.$(OBJEXT) trees.$(OBJEXT) unlzh.$(OBJEXT) \
 	unlzw.$(OBJEXT) unpack.$(OBJEXT) unzip.$(OBJEXT) \
-	util.$(OBJEXT) zip.$(OBJEXT) $(am__objects_1)
+	util.$(OBJEXT) zip.$(OBJEXT) zedc.$(OBJEXT) $(am__objects_1)
 gzip_OBJECTS = $(am_gzip_OBJECTS)
 am__DEPENDENCIES_1 =
 gzip_DEPENDENCIES = libver.a lib/libgzip.a $(am__DEPENDENCIES_1) \
@@ -299,7 +299,7 @@ am__depfiles_remade = ./$(DEPDIR)/bits.Po ./$(DEPDIR)/deflate.Po \
 	./$(DEPDIR)/inflate.Po ./$(DEPDIR)/trees.Po \
 	./$(DEPDIR)/unlzh.Po ./$(DEPDIR)/unlzw.Po \
 	./$(DEPDIR)/unpack.Po ./$(DEPDIR)/unzip.Po ./$(DEPDIR)/util.Po \
-	./$(DEPDIR)/version.Po ./$(DEPDIR)/zip.Po
+	./$(DEPDIR)/version.Po ./$(DEPDIR)/zip.Po ./$(DEPDIR)/zedc.Po
 am__mv = mv -f
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
@@ -1838,7 +1838,7 @@ top_srcdir = @top_srcdir@
 ALL_RECURSIVE_TARGETS = distcheck-hook
 BUILT_SOURCES = version.c version.h $(am__append_2) $(am__append_4) \
 	$(am__append_6)
-SUBDIRS = lib doc . tests
+SUBDIRS = lib . tests
 AM_CPPFLAGS = -I$(top_srcdir)/lib
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 
@@ -1871,7 +1871,7 @@ bin_SCRIPTS = gunzip gzexe zcat zcmp zdiff \
   zegrep zfgrep zforce zgrep $(ZLESS_PROG) zmore znew
 
 gzip_SOURCES = bits.c deflate.c gzip.c inflate.c trees.c unlzh.c \
-	unlzw.c unpack.c unzip.c util.c zip.c $(am__append_1)
+	unlzw.c unpack.c unzip.c util.c zedc.c zip.c $(am__append_1)
 gzip_LDADD = libver.a lib/libgzip.a $(CLOCK_TIME_LIB) $(FDATASYNC_LIB)
 SUFFIXES = .in
 gen_start_date = 2008-01-01
@@ -2034,6 +2034,7 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/util.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/version.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/zip.Po@am__quote@ # am--include-marker
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/zedc.Po@am__quote@ # am--include-marker
 
 $(am__depfiles_remade):
 	@$(MKDIR_P) $(@D)
@@ -2477,6 +2478,7 @@ distclean: distclean-recursive
 	-rm -f ./$(DEPDIR)/util.Po
 	-rm -f ./$(DEPDIR)/version.Po
 	-rm -f ./$(DEPDIR)/zip.Po
+	-rm -f ./$(DEPDIR)/zedc.Po
 	-rm -f Makefile
 distclean-am: clean-am distclean-compile distclean-generic \
 	distclean-tags
@@ -2538,6 +2540,7 @@ maintainer-clean: maintainer-clean-recursive
 	-rm -f ./$(DEPDIR)/util.Po
 	-rm -f ./$(DEPDIR)/version.Po
 	-rm -f ./$(DEPDIR)/zip.Po
+	-rm -f ./$(DEPDIR)/zedc.Po
 	-rm -f Makefile
 maintainer-clean-am: distclean-am maintainer-clean-generic
 
