diff --git a/unzip.c b/unzip.c
index 9880408..fb33040 100644
--- a/unzip.c
+++ b/unzip.c
@@ -133,11 +133,16 @@ unzip (int in, int out)
 
     /* Decompress */
     if (method == DEFLATED)  {
-
+	int res;
 #ifdef IBM_Z_DFLTCC
-        int res = dfltcc_inflate ();
+        res = dfltcc_inflate ();
 #else
-        int res = gzip_inflate ();
+#ifdef __MVS__
+	if (zedc)
+          res = zedc_inflate();
+        else
+#endif
+        res = gzip_inflate ();
 #endif
 
         if (res == 3) {
